### ISSUE TITLE
`evalCode` won't return error in ES module mode

### ISSUE THREAD

**Issue by yar2001** [justjake/quickjs-emscripten#160]
Thanks so much for this project!

## Bug

Starting from version `0.28.0`, when evaluating code as an ES module, `evalCode` won't return the error.

For example, the `result` won't have an `error` prop in the following code (test in browser):
```js
import { getQuickJS } from 'quickjs-emscripten';

async function main() {
  const QuickJS = await getQuickJS();
  const vm = QuickJS.newContext();

  const world = vm.newString('world');
  vm.setProp(vm.global, 'NAME', world);
  world.dispose();

  const result = vm.evalCode(`(null).abc;"Hello " + NAME + "!"`, '', {
    strict: true,
    type: 'module',
  });
  if (result.error) {
    console.log('Execution failed:', vm.dump(result.error));
    result.error.dispose();
  } else {
    console.log('Success:', vm.dump(result.value));
    result.value.dispose();
  }

  vm.dispose();
}
```

**Comment by justjake** [comment]
Does the bug reproduce with quickjs-ng variant?

Jake Teton-Landis
(805) 452-0701
***@***.***

**Comment by yar2001** [comment]
It seems that the quickjs-ng variant doesn't have this problem. `evalCode` will return the error based on the following code and the latest version `0.29.0`:
```js
import { newQuickJSWASMModuleFromVariant } from 'quickjs-emscripten';
import { useEffect } from 'react';

import releaseVariant from '@jitl/quickjs-ng-wasmfile-release-sync';

async function main() {
  const QuickJS = await newQuickJSWASMModuleFromVariant(releaseVariant);
  const vm = QuickJS.newContext();

  const world = vm.newString('world');
  vm.setProp(vm.global, 'NAME', world);
  world.dispose();

  const result = vm.evalCode(`(null).abc;"Hello " + NAME + "!"`, '', {
    strict: true,
    type: 'module',
  });
  if (result.error) {
    console.log('Execution failed:', vm.dump(result.error));
    result.error.dispose();
  } else {
    console.log('Success:', vm.dump(result.value));
    result.value.dispose();
  }

  vm.dispose();
}
```

**Comment by justjake** [comment]
This is released as of 0.29.1
